server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/NodeJoiningIT.java

```diff
@@ -220,45 +197,29 @@ Therefore, this WARN log should not be thrown (#ES-11449)
             logger.info("Sending node join request");
             String newNodeName = internalCluster().startDataOnlyNode();
 
-            safeAwait(publishingBanRemovedListener);
-            logger.info("Master publishing ban removed");
+            // Assert the master was re-elected
+            assertTrue(masterNodeName.equals(internalCluster().getMasterName()) && originalTerm < getTerm(masterNodeName));
```

Comment:
nit: slight preference for this being two `assertThat` calls: if this fails, we get no feedback about why, whereas if we use `assertThat` twice then we'll be able to see whether it was the master that changed (and it'll identify the new master) or whether the term didn't increase.